libvast/vast/system/reader_command.hpp

```diff
@@ -70,21 +71,35 @@ caf::message reader_command(const command& cmd, caf::actor_system& sys,
     else
       file = std::string{Reader::defaults::path};
   }
-  // Supply an alternate schema, if requested.
-  expected<vast::schema> schema{caf::none};
+  // Get the default schema from the registry.
+  auto schema = event_types::get();
+  // Update with an alternate schema, if requested.
+  vast::schema reader_schema;
   {
     auto sc = caf::get_if<std::string>(&options, category + ".schema");
     auto sf = caf::get_if<std::string>(&options, category + ".schema-file");
     if (sc && sf)
       return make_message(
         make_error(ec::invalid_configuration,
                    "had both schema and schema-file provided"));
-    if (sc)
-      schema = to<vast::schema>(*sc);
-    if (sf)
-      schema = load_schema_file(*sf);
-    if (!schema && schema.error() != caf::none)
-      return make_message(std::move(schema.error()));
+    auto update = [&]() -> caf::expected<vast::schema> {
+      if (sc)
+        return to<vast::schema>(*sc);
+      if (sf)
+        return load_schema_file(*sf);
+      return caf::no_error;
+    }();
```

Comment:
Mind explaining (and documenting via a comment) why you created an anonymous function and invoke it directly?

Reply:
This is to avoid pre-declaring `caf::expected<vast::schema> update;`. For me, this feels more maintainable. It's a stylistic choice, therefore I don't think it warrants a comment in the code.